scripts/transition-staging-repository-state.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+#
+# Copyright Kroxylicious Authors.
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+
+set -euo pipefail
+
+usage() {
+      1>&2 cat << EOF
+usage: $0 <-a|-s (close|drop|release)|-h>
+ -a asserts that no staging repos exist
+ -s desired state to move the staging repository (close|drop|release)
+ -h this help message
+EOF
+exit 1
+}
+
+PLUGIN=org.sonatype.plugins:nexus-staging-maven-plugin:1.6.13
+MVN_ARGS=("--batch-mode" "--no-transfer-progress" "-DnexusUrl=https://s01.oss.sonatype.org/" "-DserverId=ossrh")
+
+ASSERT_NO_STAGING_REPOS="false"
+STATE=""
+while getopts ":s:ah" opt; do
+  case $opt in
+    a) ASSERT_NO_STAGING_REPOS="true"
+    ;;
+    s) STATE="${OPTARG}"
+    ;;
+    h) usage
+    ;;
+    \?) echo "Invalid option -${OPTARG}" >&2
+    exit 1
+    ;;
+  esac
+done
+
+# The rc-list table looks like this:
+#[INFO] ID                   State    Description
+#[INFO] iokroxylicious-1032  CLOSED   unknown
+#[INFO] ------------------------------------------------------------------------
+
+STAGING_REPO_IDS=$(mvn ${PLUGIN}:rc-list "${MVN_ARGS[@]}" 2>/dev/null \
+                 | awk '/\[INFO\] ---*/{found=0} found==1{print $2} /\[INFO\] ID.*State.*Description/{found=1}')
```

Comment:
Yuck. thats a horrible thing to need.

Reply:
Indeed, I couldn't believe the maven plugin doesn't give a programmatic way to retrieve the information.   I think they are assuming you stage the release using their plugin which gives you control of the id, so you never need to discover it.